theano/gpuarray/opt.py

```diff
@@ -845,10 +848,14 @@ def local_gpu_pdbbreakpoint_op(node):
 def local_gpua_lazy_ifelse(op, context_name, inputs):
     if op.gpu:
         return
+    if isinstance(inputs[0].type, GpuArrayType):
+        return
     c = inputs[0]
     inps = []
     for v in inputs[1:]:
-        if isinstance(v.type, (tensor.TensorType, GpuArrayType)):
+        if isinstance(v.type, GpuArrayType):
+            return
+        elif isinstance(v.type, tensor.TensorType):
```

Comment:
I don't understand why this change fix something. I would think the old code was good. Can you explain?